### ISSUE TITLE
 build fails for iOS on Ionic Pro due to duplicate symbol _OBJC_CLASS_$_CDVLogger

### ISSUE THREAD

**Issue by alexa2lego** [ionic-team/ionic#12852]
[31m[x] duplicate symbol _OBJC_CLASS_$_CDVLogger in[0m
> CDVLogger.o
> libCordova.a(CDVLogger.o)

[31m[x] duplicate symbol _OBJC_METACLASS_$_CDVLogger in[0m
> CDVLogger.o
> libCordova.a(CDVLogger.o)

[31m[x] ld: 2 duplicate symbols for architecture armv7[0m

the same build worked 10 days ago  and today the build does not work more.

**Ionic version:**  (check one with "x")
[ ] **1.x** (For Ionic 1.x issues, please use https://github.com/ionic-team/ionic-v1)
[x] **2.x**
[ ] **3.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**
build iOS on ionic pro

**Expected behavior:**
build new version in dashboard.ionicjs.com

**Steps to reproduce:**
Application Id: 5a165360

[0KRunning with gitlab-ci-multi-runner 9.1.0 (0118d89)
  on Mac OSX Runner (9327ea78)
[0;m[0KUsing Shell executor...
[0;mRunning on Ionics-Mac-3.local...
[32;1mCloning repository...[0;m
Cloning into '/Users/ionic/builds/9327ea78/0/project-1'...
[32;1mChecking out a0233bad as master...[0;m
[32;1mSkipping Git submodules setup[0;m
[32;1m$ fetch-updates[0;m
Checking for build process updates...
[32;1m$ build-ios[0;m
Downloading cached www/ dir...
Successfully downloaded www/ dir...
Installing cordova@7.0.1
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
/usr/local/bin/cordova -> /usr/local/lib/node_modules/cordova/bin/cordova
- json-stable-stringify@1.0.1 node_modules/cordova/node_modules/ajv/node_modules/json-stable-stringify
- readable-stream@1.0.34 node_modules/cordova/node_modules/bl/node_modules/readable-stream
- bl@0.9.5 node_modules/cordova/node_modules/bl
- co@4.6.0 node_modules/cordova/node_modules/co
- ajv@4.11.8 node_modules/cordova/node_modules/ajv
- underscore@1.8.3 node_modules/cordova/node_modules/cordova-common/node_modules/underscore
- q@1.5.0 node_modules/cordova/node_modules/cordova-create/node_modules/cordova-common/node_modules/q
- shelljs@0.5.3 node_modules/cordova/node_modules/cordova-create/node_modules/cordova-common/node_modules/shelljs
- sax@0.3.5 node_modules/cordova/node_modules/cordova-create/node_modules/sax
- elementtree@0.1.6 node_modules/cordova/node_modules/cordova-create/node_modules/elementtree
- underscore@1.8.3 node_modules/cordova/node_modules/cordova-create/node_modules/underscore
- cordova-common@2.0.3 node_modules/cordova/node_modules/cordova-create/node_modules/cordova-common
- shelljs@0.5.3 node_modules/cordova/node_modules/cordova-fetch/node_modules/cordova-common/node_modules/shelljs
glob@7.1.2 node_modules/cordova/node_modules/cordova-fetch/node_modules/shelljs/node_modules/glob -> node_modules/cordova/node_modules/cordova-fetch/node_modules/glob
- sax@0.3.5 node_modules/cordova/node_modules/cordova-fetch/node_modules/sax
- elementtree@0.1.6 node_modules/cordova/node_modules/cordova-fetch/node_modules/elementtree
- underscore@1.8.3 node_modules/cordova/node_modules/cordova-fetch/node_modules/underscore
- cordova-common@2.0.3 node_modules/cordova/node_modules/cordova-fetch/node_modules/cordova-common
- sax@0.3.5 node_modules/cordova/node_modules/cordova-lib/node_modules/sax
- elementtree@0.1.6 node_modules/cordova/node_modules/cordova-lib/node_modules/elementtree
- unorm@1.3.3 node_modules/cordova/node_modules/cordova-lib/node_modules/unorm
- ctype@0.5.3 node_modules/cordova/node_modules/ctype
- once@1.3.3 node_modules/cordova/node_modules/end-of-stream/node_modules/once
- ms@2.0.0 node_modules/cordova/node_modules/express/node_modules/ms
- debug@2.6.7 node_modules/cordova/node_modules/express/node_modules/debug
- ms@2.0.0 node_modules/cordova/node_modules/finalhandler/node_modules/ms
- debug@2.6.7 node_modules/cordova/node_modules/finalhandler/node_modules/debug
- mime@1.2.11 node_modules/cordova/node_modules/form-data/node_modules/mime
- har-schema@1.0.5 node_modules/cordova/node_modules/har-schema
- assert-plus@0.2.0 node_modules/cordova/node_modules/insight/node_modules/assert-plus
- async@1.5.2 node_modules/cordova/node_modules/insight/node_modules/async
- aws-sign2@0.6.0 node_modules/cordova/node_modules/insight/node_modules/aws-sign2
- caseless@0.12.0 node_modules/cordova/node_modules/insight/node_modules/caseless
- delayed-stream@1.0.0 node_modules/cordova/node_modules/insight/node_modules/delayed-stream
- combined-stream@1.0.5 node_modules/cordova/node_modules/insight/node_modules/combined-stream
- forever-agent@0.6.1 node_modules/cordova/node_modules/insight/node_modules/forever-agent
- form-data@2.1.4 node_modules/cordova/node_modules/insight/node_modules/form-data
- hoek@2.16.3 node_modules/cordova/node_modules/insight/node_modules/hoek
- boom@2.10.1 node_modules/cordova/node_modules/insight/node_modules/boom
- cryptiles@2.0.5 node_modules/cordova/node_modules/insight/node_modules/cryptiles
...

[33m[!] [0m/Users/ionic/builds/9327ea78/0/project-1/platforms/ios/SchoolAdvisor/Plugins/cordova-plugin-ionic/UNIHTTPClientHelper.m:127:101: [33mvalues of type 'NSUInteger' should not be used as format arguments; add an explicit cast to 'unsigned long' instead [-Wformat][0m

[body appendData:[[NSString stringWithFormat:@"Content-Length: %d\r\n\r\n", data.length] dataUsingEncoding:NSUTF8StringEncoding]];
[36m            ^[0m

[33m[!] [0m/Users/ionic/builds/9327ea78/0/project-1/platforms/ios/SchoolAdvisor/Plugins/cordova-plugin-ionic/UNIHTTPClientHelper.m:206:38: [33m'sendSynchronousRequest:returningResponse:error:' is deprecated: first deprecated in iOS 9.0 - Use [NSURLSession dataTaskWithRequest:completionHandler:] (see NSURLSession.h [-Wdeprecated-declarations][0m

NSData * data = [NSURLConnection sendSynchronousRequest:requestObj returningResponse:&response error:error];
[36m                                                                                       ~~           ^~~~~~~~~~~[0m

[33m>[0m [39;1mCompiling[0m UNIHTTPRequest.m
[33m>[0m [39;1mCompiling[0m UNIHTTPRequestWithBody.m
[33m>[0m [39;1mCompiling[0m UNIRest.m
[33m>[0m [39;1mCompiling[0m UNIUrlConnection.m

[33m[!] [0m/Users/ionic/builds/9327ea78/0/project-1/platforms/ios/SchoolAdvisor/Plugins/cordova-plugin-ionic/UNIUrlConnection.m:53:43: [33m'initWithRequest:delegate:startImmediately:' is deprecated: first deprecated in iOS 9.0 - Use NSURLSession (see NSURLSession.h) [-Wdeprecated-declarations][0m

connection = [[NSURLConnection alloc] initWithRequest:request delegate:self startImmediately:NO];
[36m^[0m

[33m>[0m [39;1mCompiling[0m UNIBaseRequest.m
[33m>[0m [39;1mCompiling[0m UNIBodyRequest.m
[33m>[0m [39;1mCompiling[0m UNISimpleRequest.m
[33m>[0m [39;1mCompiling[0m UNIHTTPBinaryResponse.m
[33m>[0m [39;1mCompiling[0m UNIHTTPJsonResponse.m
[33m>[0m [39;1mCompiling[0m UNIHTTPResponse.m
[33m>[0m [39;1mCompiling[0m UNIHTTPStringResponse.m
[33m>[0m [39;1mCompiling[0m UNIJsonNode.m
[33m>[0m [39;1mCompiling[0m CDVSplashScreen.m
[33m>[0m [39;1mCompiling[0m CDVViewController+SplashScreen.m
[33m>[0m [39;1mCompiling[0m CDVStatusBar.m

[33m[!] [0m/Users/ionic/builds/9327ea78/0/project-1/platforms/ios/SchoolAdvisor/Plugins/cordova-plugin-statusbar/CDVStatusBar.m:281:44: [33m'setStatusBarStyle:' is deprecated: first deprecated in iOS 9.0 - Use -[UIViewController preferredStatusBarStyle] [-Wdeprecated-declarations][0m

[[UIApplication sharedApplication] setStatusBarStyle:style];
[36m^[0m

[33m[!] [0m/Users/ionic/builds/9327ea78/0/project-1/platforms/ios/SchoolAdvisor/Plugins/cordova-plugin-statusbar/CDVStatusBar.m:378:14: [33m'setStatusBarHidden:' is deprecated: first deprecated in iOS 9.0 - Use -[UIViewController prefersStatusBarHidden] [-Wdeprecated-declarations][0m

[app setStatusBarHidden:YES];
[36m                                                 ^[0m

[33m[!] [0m/Users/ionic/builds/9327ea78/0/project-1/platforms/ios/SchoolAdvisor/Plugins/cordova-plugin-statusbar/CDVStatusBar.m:411:14: [33m'setStatusBarHidden:' is deprecated: first deprecated in iOS 9.0 - Use -[UIViewController prefersStatusBarHidden] [-Wdeprecated-declarations][0m

[app setStatusBarHidden:NO];
[36m                                                              ^[0m

[33m[!] [0m/Users/ionic/builds/9327ea78/0/project-1/platforms/ios/SchoolAdvisor/Plugins/cordova-plugin-statusbar/CDVStatusBar.m:481:48: [33m'applicationFrame' is deprecated: first deprecated in iOS 9.0 - Use -[UIScreen bounds] [-Wdeprecated-declarations][0m

CGRect bounds = [[UIScreen mainScreen] applicationFrame];
[36m                                                              ^[0m

[33m>[0m [39;1mCompiling[0m IonicKeyboard.m
[33m>[0m [39;1mLinking[0m SchoolAdvisor

[31m[x] duplicate symbol _OBJC_CLASS_$_CDVLogger in[0m
> CDVLogger.o
> libCordova.a(CDVLogger.o)

[31m[x] duplicate symbol _OBJC_METACLASS_$_CDVLogger in[0m
> CDVLogger.o
> libCordova.a(CDVLogger.o)

[31m[x] ld: 2 duplicate symbols for architecture armv7[0m

[31m[x] clang: error: linker command failed with exit code 1 (use -v to see invocation)[0m

[33m>[0m [39;1mLinking[0m SchoolAdvisor
** ARCHIVE FAILED **

The following build commands failed:
Ld /Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/IntermediateBuildFilesPath/SchoolAdvisor.build/Release-iphoneos/SchoolAdvisor.build/Objects-normal/armv7/SchoolAdvisor normal armv7
(1 failure)
[11:54:09]: [31mExit status: 65[0m

+-------------+-------------------------+
|           [32m[33mBuild environment[0m           |
+-------------+-------------------------+
| xcode_path  | /Applications/Xcode.app |
| gym_version | 2.46.1                  |
| sdk         | iPhoneOS10.3.sdk        |
+-------------+-------------------------+

[11:54:09]: ▸ [35mLd /Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/IntermediateBuildFilesPath/SchoolAdvisor.build/Release-iphoneos/SchoolAdvisor.build/Objects-normal/arm64/SchoolAdvisor normal arm64[0m
[11:54:09]: ▸ [35m    cd /Users/ionic/builds/9327ea78/0/project-1/platforms/ios[0m
[11:54:09]: ▸ [35m    export IPHONEOS_DEPLOYMENT_TARGET=9.0[0m
[11:54:09]: ▸ [35m    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"[0m
[11:54:09]: ▸ [35m    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.3.sdk -L/Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/BuildProductsPath/Release-iphoneos -F/Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/BuildProductsPath/Release-iphoneos -filelist /Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/IntermediateBuildFilesPath/SchoolAdvisor.build/Release-iphoneos/SchoolAdvisor.build/Objects-normal/arm64/SchoolAdvisor.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -miphoneos-version-min=9.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/IntermediateBuildFilesPath/SchoolAdvisor.build/Release-iphoneos/SchoolAdvisor.build/Objects-normal/arm64/SchoolAdvisor_lto.o -Xlinker -final_output -Xlinker /Applications/SchoolAdvisor.app/SchoolAdvisor -fobjc-arc -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos -Xlinker -add_ast_path -Xlinker /Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/IntermediateBuildFilesPath/SchoolAdvisor.build/Release-iphoneos/SchoolAdvisor.build/Objects-normal/arm64/SchoolAdvisor.swiftmodule -ObjC /Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/BuildProductsPath/Release-iphoneos/libCordova.a -lz -Xlinker -dependency_info -Xlinker /Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/IntermediateBuildFilesPath/SchoolAdvisor.build/Release-iphoneos/SchoolAdvisor.build/Objects-normal/arm64/SchoolAdvisor_dependency_info.dat -o /Users/ionic/Library/Developer/Xcode/DerivedData/SchoolAdvisor-gizbdzbxoankgbdbbfxamesrohfc/Build/Intermediates/ArchiveIntermediates/SchoolAdvisor/IntermediateBuildFilesPath/SchoolAdvisor.build/Release-iphoneos/SchoolAdvisor.build/Objects-normal/arm64/SchoolAdvisor[0m
[11:54:09]: 
[11:54:09]: [31m⬆️  Check out the few lines of raw `xcodebuild` output above for potential hints on how to solve this error[0m
[11:54:09]: [33m📋  For the complete and more detailed error log, check the full log at:[0m
[11:54:09]: [33m📋  /Users/ionic/Library/Logs/gym/SchoolAdvisor-SchoolAdvisor.log[0m
[11:54:09]: 
[11:54:09]: [31mLooks like fastlane ran into a build/archive error with your project[0m
[11:54:09]: [31mIt's hard to tell what's causing the error, so we wrote some guides on how[0m
[11:54:09]: [31mto troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/[0m
[11:54:09]: [31mBefore submitting an issue on GitHub, please follow the guide above and make[0m
[11:54:09]: [31msure your project is set up correctly.[0m
[11:54:09]: [31mfastlane uses `xcodebuild` commands to generate your binary, you can see the[0m
[11:54:09]: [31mthe full commands printed out in yellow in the above log.[0m
[11:54:09]: [31mMake sure to inspect the output above, as usually you'll find more error information there[0m
[11:54:09]: 
[11:54:09]: [32m---------------------------[0m
[11:54:09]: [32m--- Step: shell command ---[0m
[11:54:09]: [32m---------------------------[0m
[11:54:09]: [32m-----------------------------[0m
[11:54:09]: [32m--- Step: delete_keychain ---[0m
[11:54:09]: [32m-----------------------------[0m
[11:54:09]: Error building the application - see the log above
+-------------------------------------+--------------------------------------+
|                                [33mLane Context[0m                                |
+-------------------------------------+--------------------------------------+
| DEFAULT_PLATFORM                    | ios                                  |
| PLATFORM_NAME                       | ios                                  |
| LANE_NAME                           | ios package_build                    |
| DOWNLOAD_CERTS_CODESIGNING_IDENTITY | iPhone Distribution: Vadim Kantor    |
|                                     | (532YXFZLRD)                         |
| DOWNLOAD_CERTS_TEAM_ID              | 532YXFZLRD                           |
| DOWNLOAD_CERTS_PROJECT_NAME         | SchoolAdvisor                        |
| DOWNLOAD_CERTS_CERT_PATH            | /Users/ionic/builds/9327ea78/0/proj  |
|                                     | ect-1/cert_file.p12                  |
| DOWNLOAD_CERTS_PROVISIONING_PATH    | /Users/ionic/builds/9327ea78/0/proj  |
|                                     | ect-1/provision_file.mobileprovisio  |
|                                     | n                                    |
+-------------------------------------+--------------------------------------+
[11:54:09]: [31mError building the application - see the log above[0m

+------+------------------------+-------------+
|              [32mfastlane summary[0m               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | default_platform       | 0           |
| 2    | create_keychain        | 0           |
| 3    | download_certs         | 0           |
| 4    | import_certificate     | 0           |
| 5    | update_project_codesi  | 0           |
|      | gning                  |             |
| 6    | update_project_provis  | 0           |
|      | ioning                 |             |
| 7    | shell command          | 0           |
| 💥   | [31mgym[0m                    | 50          |
| 9    | shell command          | 0           |
| 10   | delete_keychain        | 0           |
+------+------------------------+-------------+

[11:54:09]: [31mfastlane finished with errors[0m
[31m
[!] Error building the application - see the log above[0m
[32;1mRunning after script...[0;m
[32;1m$ clean-up[0;m
Cleaning up files...
Successful clean up
[31;1mERROR: Job failed: exit status 1
[0;m

**Comment by dhavalsoni2001** [comment]
Facing same issue.
```
duplicate symbol _OBJC_CLASS_$_CDVLogger in:
    /Users/dhavalsoni/Library/Developer/Xcode/DerivedData/myProj-clzckrqggoyxvzfaxuwqlbatkprj/Build/Intermediates/myProj.build/Debug-iphoneos/myProj.build/Objects-normal/arm64/CDVLogger.o
    /Users/dhavalsoni/Library/Developer/Xcode/DerivedData/myProj-clzckrqggoyxvzfaxuwqlbatkprj/Build/Products/Debug-iphoneos/libCordova.a(CDVLogger.o)
duplicate symbol _OBJC_METACLASS_$_CDVLogger in:
    /Users/dhavalsoni/Library/Developer/Xcode/DerivedData/myProj-clzckrqggoyxvzfaxuwqlbatkprj/Build/Intermediates/myProj.build/Debug-iphoneos/myProj.build/Objects-normal/arm64/CDVLogger.o
    /Users/dhavalsoni/Library/Developer/Xcode/DerivedData/myProj-clzckrqggoyxvzfaxuwqlbatkprj/Build/Products/Debug-iphoneos/libCordova.a(CDVLogger.o)
ld: 2 duplicate symbols for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by BhawaniSingh** [comment]
Duplicate of https://github.com/ionic-team/ionic/issues/12849

Solution: https://github.com/ionic-team/ionic/issues/12849#issuecomment-328472880

**Comment by kensodemann** [comment]
Hello! Thank you for opening an issue with us! I am closing this issue because it is a duplicate of Issue #12849 as already mentioned. Thank you for using Ionic

**Comment by paqogomez** [comment]
The actual solution (for me at least) was to remove the cordova-plugin-console as noted a bit further down the thread that @BhawaniSingh mentions here:

https://github.com/ionic-team/ionic/issues/12849#issuecomment-331105393

**Comment by laberg** [comment]
Thanks a lot @paqogomez

**Comment by FieldPromax** [comment]
got this solution after spending two days of resolving this build issue. **THANKS**

**Comment by WhiteVulpine** [comment]
@paqogomez YES! Thank you! Finally after much troubleshooting and many issues I built successfully

Notes for my iOS build with push plugins to succeed

* Downgrade phonegap-plugin-push (I was using version 2.2.2)
  * `cordova plugin rm phonegap-plugin-push`
  * `cordova plugin add phonegap-plugin-push@2.1.2`
* Missing library GoogleToolBox and other library issues. Install CocoaPods
  * `cd platform/ios; pod setup; pod repo update; pod install;`
  * If you are getting hung up on the dependency process, download/update git to the most recent version. I updated from 2.15.4 to 2.17.0 and that resolved this issue
